### PR TITLE
Resource I/O and dynamic content visualization

### PR DESCRIPTION
**Pull request by BelfDev**
# Resource I/O & Dynamic Content Visualization

- Adds a subtitle to the `NavigationBar`
- Creates a generic CSV parsing and writing logic
- Adds [OpenCSV](http://opencsv.sourceforge.net/) dependency
- Creates the  `ApplicationRepository` to handle data
- Introduces a `JTable` and `JScrollPane` in `CoursePlannerScreen`
- Creates a dynamic content visualization logic that takes into account the user `Role` to present the table data

*With this PR the following stories are complete:*
- [x] [As a Class Director, I want to see the list of courses and corresponding teaching requirements (2 pt.)](https://trello.com/c/aEVti9Kp)
- [x] [As an Administrator, I want to see the list of teaching requirements, a corresponding course and assigned staff for each requirement (1 pt.)](https://trello.com/c/s4lNT83q)
- [x] [As a PTT Director, I want to see the list of courses and corresponding teaching requirements with their status (approved, not approved) (1 pt.)](https://trello.com/c/DKgCsux9)

### CODE REVIEW

File: src/main/java/madstax/model/util/ResourceIO.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+        URL csvURL = getResourceURL(csvFileRelativePath);
+
+        try {
+            Path filePath = Paths.get(csvURL.toURI());
+            Reader reader = Files.newBufferedReader(filePath);
+
+            HeaderColumnNameMappingStrategy<T> strategy = new HeaderColumnNameMappingStrategy<T>();
+            strategy.setType(type);
+
+            List<T> data = new CsvToBeanBuilder<T>(reader)
```
Comment (rjs68) [inline src/main/java/madstax/model/util/ResourceIO.java:44]: Beans!!!

File: src/main/java/madstax/view/NavigationBar.java [hunk 2]
```diff
@@ -53,7 +54,7 @@ private void createNavBarDefaultItems() {
         backButton.setAlignmentX(SwingConstants.LEFT);
         backButton.setVisible(false);
 
-        JLabel titleLabel = new JLabel("Role Selection");
+        JLabel titleLabel = new JLabel("Title");
```
Comment (rjs68) [inline src/main/java/madstax/view/NavigationBar.java]: Is it worth making "Title" a constant like you did within the CoursePlannerScreen?

File: src/main/resources/csv/course-plan.csv [hunk 3]
```diff
@@ -0,0 +1,5 @@
+INDEX,COURSE,REQUIREMENTS,TEACHERID,STATUS
+1,Engineering, "Maths, Physics",,UNASSIGNED
```
Comment (rjs68) [inline src/main/resources/csv/course-plan.csv:2]: Assign it to me! I'm sick at engineering

File: src/main/resources/csv/teachers.csv [hunk 4]
```diff
@@ -0,0 +1,5 @@
+ID,NAME,QUALIFICATIONS
+1,John Smith,"Maths,Physics, Chemestry"
```
Comment (rjs68) [inline src/main/resources/csv/teachers.csv:2]: This should say Rhys - I have all these A levels. Deeply offended you did not think of me

File: src/main/resources/csv/teachers.csv [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+2,Lily Allen,"Arts, Dancing"
+3,Walt Disney,"Drawing, Animation"
+4,Nathan,"Arts, Filming"
```
Comment (rjs68) [inline src/main/resources/csv/teachers.csv:5]: Not certain he actually has these skills

File: src/test/java/madstax/LauncherTest.java [hunk 6]
```diff
@@ -1,20 +1,18 @@
...
+public class LauncherTest {
     /**
      * Rigorous Test :-)
      */
     @Test
-    public void shouldAnswerWithTrue()
-    {
-        assertTrue( true );
+    public void shouldAnswerWithTrue() {
+        assertTrue(true);
```
Comment (rjs68) [inline src/test/java/madstax/LauncherTest.java:16]: Very extensive testing - well done!

**Review by rjs68 — APPROVED** [review]
Beautiful work as always!

File: src/main/java/madstax/model/util/ResourceIO.java (hunk 1 above)
Comment (BelfDev) [reply src/main/java/madstax/model/util/ResourceIO.java:44]: hahahah

File: src/main/java/madstax/view/NavigationBar.java (hunk 2 above)
Comment (BelfDev) [reply src/main/java/madstax/view/NavigationBar.java]: Removed!

**Review by nathanieldschneider — APPROVED** [review]
beans

**Pull request merged**